### PR TITLE
add validation of MLM output using classification:classes (fixes #48)

### LINKED ISSUE #48: Proposal: Extend the Classification schema for integration with Machine Learning Model extension
For the MLM (Machine Learning Model) extension, we are planing to reuse the definition of https://github.com/stac-extensions/classification?tab=readme-ov-file#class-object within Model Output definitions, since they are extremely related. For example, the following `mlm:ouput` can directly take advantage of `classification:classes` to indicate which class pixel values are expected to be produced by the model. Using this definition, creating a derived STAC Item product from the model prediction is simplified, since one only needs to copy the nested `classification:classes` directly into the derived STAC Item.

```json
{
  "properties": {
    "mlm:output": [
      {
        "tasks": [ "segmentation", "semantic-segmentation" ],
        "result": {
          "shape": [ -1, 10980, 10980 ],
          "dim_names": [ "CLASSIFICATION" ],
          "data_type": "float32"
        },
        "post_processing_function": null,
        "classification:classes": [
          {
            "name": "WATER",
            "value": 1,
            "color_hint": [ 0, 0, 255 ],
            "nodata": false
          },
          {
            "name": "NON-WATER",
            "value": 0,
            "color_hint": [ 0, 0, 0 ],
            "nodata": false
          }
        ]
      }
    ]
  }
}
```

For the moment, the MLM JSON schema can point directly at the `classification:classes` definition to fulfil the validation, but the `stac_extensions` of the MLM STAC Item cannot really include the `classification` schema, since no field is technically respecting it at the moment. Would it be possible to do something similar to the strategy employed for `raster:bands`/`classification:classes` below?

https://github.com/stac-extensions/classification/blob/a1e713978b8cd9ac3c8b4778d140179d5e8c9022/json-schema/schema.json#L225-L236

https://github.com/stac-extensions/classification/blob/a1e713978b8cd9ac3c8b4778d140179d5e8c9022/examples/item-classes-maxar.json#L146-L148

### PR DESCRIPTION
**Pull request by fmigneault**
As proposed in #48, adds the validation of `classification:classes` nested under an `mlm:output` definition.

Adds an example demonstrating how it can be used.

~For the moment, the MLM URI validation fails inside the npm checker as it is not yet published on stac-extensions (to be discussed in the next community meeting how to make that happen). The corresponding Item (https://github.com/crim-ca/dlm-extension/blob/v1.0.0/examples/item_raster_bands.json) is however validated in using the mapping in the repository https://github.com/crim-ca/dlm-extension/blob/v1.0.0/package.json#L8-L9.~

Uses the URI defined on https://crim-ca.github.io/mlm-extension/v1.1.0/schema.json

Looking for initial feedback in case some adjustments must be done before the MLM extension is published officially.

Fixes #48

### CODE REVIEW

**Review by m-mohr — APPROVED** [review]

**Comment by fmigneault** [comment]
@m-mohr Should we wait more for feedback or can it be merged as is?

**Comment by m-mohr** [comment]
I think we can merge, but was just thinking that we should add a note in the Readme and a change log entry.

**Comment by fmigneault** [comment]
@m-mohr Added in the latest commits.

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- Validation by the JSON schema of `classification:classes` definition within a
+  [Machine Learning Model (MLM)](https://github.com/crim-ca/mlm-extension) output.
+- Example STAC `item-model-classes.json` demonstrating the use of `classification:classes` within a `mlm:output`.
+- `nodata` to `classification:classes`
 - `nodata` to Class Object
 - `percentage` to Class Object
 - `count` to Class Object
```
Comment (m-mohr) [inline CHANGELOG.md:18]:
```suggestion
- `nodata` to the Class Object
- `nodata` to the Class Object
- `percentage` to the Class Object
- `count` to the Class Object
```

**Comment by m-mohr** [comment]
> @m-mohr Added in the latest commits.

With regards to the README, I think it should be mentioned here:
https://github.com/stac-extensions/classification/pull/50/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R44-R51

Otherwise it's not clear from the README directly that it can be used in MLM.

**Comment by fmigneault** [comment]
@m-mohr Added in [c10239a](https://github.com/stac-extensions/classification/pull/50/commits/c10239a55e6850676228f71a80b89225fe3572e2)

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- Validation by the JSON schema of `classification:classes` definition within a
+  [Machine Learning Model (MLM)](https://github.com/crim-ca/mlm-extension) output.
+- Example STAC `item-model-classes.json` demonstrating the use of `classification:classes` within a `mlm:output`.
+- `nodata` to `classification:classes`
```
Comment (m-mohr) [inline CHANGELOG.md:15]:
```suggestion
- `nodata` to Class Object
```

**Comment by m-mohr** [comment]
Thanks.

**Pull request merged**